src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,233 @@
+                        = connection.prepareStatement(addDriverRelationQuery)) {
+            addDriverRelationStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                addDriverRelationStatement.setLong(2, driver.getId());
+                addDriverRelationStatement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add drivers relations for car with id "
+                    + car.getId(), e);
+        }
+    }
+
+    private void removeDriversRelations(Car car) {
+        String removeDriversRelationsQuery = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement removeDriversRelationsStatement
+                        = connection.prepareStatement(removeDriversRelationsQuery)) {
+            removeDriversRelationsStatement.setLong(1, car.getId());
+            removeDriversRelationsStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete drivers relations for car with id "
+                    + car.getId(), e);
+        }
+    }
+
+    private List<Driver> getDriversForCar(Long carId) {
+        String getDriversForCarQuery = "SELECT d.id as driver_id, d.name as driver_name, "
+                + "d.license_number FROM drivers d "
+                + "JOIN cars_drivers cd ON d.id = cd.driver_id WHERE cd.car_id = ? "
+                + "AND d.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getDriversForCarStatement
+                        = connection.prepareStatement(getDriversForCarQuery)) {
+            getDriversForCarStatement.setLong(1, carId);
+            ResultSet resultSet = getDriversForCarStatement.executeQuery();
+            List<Driver> drivers = new ArrayList<>();
+            while (resultSet.next()) {
+                drivers.add(parseDriversFromResultSet(resultSet));
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get drivers for car with id " + carId,
+                    e);
+        }
+    }
+
+    private Car getCarWithManufacturerFromResultSet(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setModel(resultSet.getString("car_model"));
+        car.setId(resultSet.getObject("car_id", Long.class));
+        Manufacturer manufacturer
+                = new Manufacturer(resultSet.getString("manufacturer_name"),
+                resultSet.getString("manufacturer_country"));
+        manufacturer.setId(resultSet.getObject("manufacturer_id", Long.class));
+        car.setManufacturer(manufacturer);
+        return car;
+    }
+
+    private void parseDriversToListOfCars(List<Car> cars) {
+        if (cars.size() != 0) {
```

Comment:
this check is redundant if size is 0 we won't enter the loop

Reply:
done